accesshandler/pkg/providers/aws/flask/validate.go

```diff
@@ -0,0 +1,63 @@
+package flask
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/identitystore"
+	"github.com/aws/aws-sdk-go-v2/service/organizations"
+	orgtypes "github.com/aws/aws-sdk-go-v2/service/organizations/types"
+	"github.com/common-fate/granted-approvals/accesshandler/pkg/diagnostics"
+	"github.com/common-fate/granted-approvals/accesshandler/pkg/providers"
+)
+
+// Validate the access against AWS SSO without actually granting it.
+// This provider requires that the user name matches the user's email address.
+
+func (p *Provider) ensureAccountExists(ctx context.Context, accountID string) error {
+	_, err := p.orgClient.DescribeAccount(ctx, &organizations.DescribeAccountInput{
+		AccountId: &accountID,
+	})
+	var anf *orgtypes.AccountNotFoundException
+	if errors.As(err, &anf) {
+		return &AccountNotFoundError{AccountID: accountID}
+	}
+
+	return err
+}
+func (p *Provider) ValidateConfig() map[string]providers.ConfigValidationStep {
+	return map[string]providers.ConfigValidationStep{
+		"sso-list-users": {
+			Name:            "List users in the AWS SSO instance",
+			FieldsValidated: []string{"instanceArn", "region", "identityStoreId"},
+			Run: func(ctx context.Context) diagnostics.Logs {
+				// try and list users in the AWS SSO instance.
+				res, err := p.idStoreClient.ListUsers(ctx, &identitystore.ListUsersInput{
+					IdentityStoreId: aws.String(p.identityStoreID.Get()),
+				})
+				if err != nil {
+					return diagnostics.Error(err)
+				}
+				return diagnostics.Info("AWS SSO returned %d users (more may exist, pagination has been ignored)", len(res.Users))
+			},
+		},
+		"assume-role": {
```

Comment:
This needs to be replaced with a test that actually tries to assume a role - see the AWS SSO v2 provider as this has now been fixed there.